mesonbuild/modules/simd.py

```diff
@@ -75,39 +75,52 @@ def __init__(self, interpreter: Interpreter):
                   *[SOURCES_KW.evolve(name=iset) for iset in ISETS],
                   allow_unknown=True) # Because we also accept STATIC_LIB_KWS, but we check them in the interpreter call later on.
     def check(self, state: ModuleState, args: T.Tuple[str], kwargs: CheckKw) -> T.List[T.Union[T.List[build.StaticLibrary], build.ConfigurationData]]:
-        result: T.List[build.StaticLibrary] = []
         if 'sources' in kwargs:
             raise mesonlib.MesonException('SIMD module does not support the "sources" keyword')
+
+        result: T.List[build.StaticLibrary] = []
         prefix = args[0]
-        basic_kwargs = {}
-        for key, value in kwargs.items():
-            if key not in ISETS and key != 'compiler':
-                basic_kwargs[key] = value
         compiler = kwargs['compiler']
         conf = build.ConfigurationData()
+
+        static_lib_kwargs = T.cast('T.Dict[str, T.Any]', kwargs.copy())
+        del static_lib_kwargs['compiler']
         for iset in ISETS:
-            if iset not in kwargs:
+            del static_lib_kwargs[iset]
+        static_lib_kwargs = T.cast('kwtypes.StaticLibrary', static_lib_kwargs) # type: ignore
```

Comment:
I think this could just be written as:
```python
local_keys = set(ISETS) + {'compiler'}
static_lib_kwargs = T.cast('kwtypes.StaticLibrary', {k: v for k, v in kwargs.items() if k not in local_keys))
```
which avoids the full copy with deletions, as well as the need to recast